### ISSUE TITLE
Remove container if single user server does not start

### ISSUE THREAD

**Issue by kazeevn** [everware/everware#91]
Hi,

I've just tried to start Everware for my repository https://github.com/yandexdataschool/reproducible_analysis_course_py3 . It failed:

> ```
> Cloning repository https://github.com/yandexdataschool/reproducible_analysis_course_py3.git
> Building image (everware/https_github_com_yandexdataschool_reproducible_analysis_course_py3_git-bf46684710e69ea9d0e9f88548738c6539dd4c5b)
> Creating container
> Something went wrong during building. Error: APIError(HTTPError('409 Client Error: Conflict for url: http://fol6-jupyterhub01.mlhep.yandex.net:31338/v1.21/containers/create?name=jupyter-kazeevn',),)
> ```

**Comment by anaderi** [comment]
it should be fixed in #79

**Comment by betatim** [comment]
Do you know what the error is/was?

**Comment by anaderi** [comment]
now if there is no jupyter instance started within container, container is still created, but everware cannot connect to it, and fails robustly (but silently). next time user attempts to create the container, docker fails to do so, since one is already there and says 'Conflict for url'

**Comment by betatim** [comment]
Ok. The correct behaviour would be to delete the container if the singleuser server can't be reached. Should we upstream this behaviour to `dockerspawner`?

**Comment by anaderi** [comment]
AFAIK, in #79 we actually rely on some `dockerspawner` defaults. so there is no much to upstream. 
/cc @astiunov ?

**Comment by astiunov** [comment]
`dockerspawner` doesn't include any errors' handling, so if the existing container is incorrect, `dockerspawner` just tries to start it and fails. But after that `jupyterhub` itself should call `stop` method of `dockerspawner`, which removes this container.

**Comment by betatim** [comment]
Yeah, dealing with a spawner that fails to start (in this case because no server comes up) seems like something that should be dealt with in `jupyterhub` or `dockerspawner`. Probably best handled in a spawner as jupyterhub can also use local processes as spawner and if they don't show up there isn't much clean up to do.

Detecting that something went wrong with the container and clean it up should be the job of `dockerspawner`.

**Comment by betatim** [comment]
My suggestion would be to create an issue in `jupyter/dockerspawner` to discuss the option of the spawner detecting that the server did not come up and removing the container again.
